File: backend/src/algoritmos/sort/bubble_sort/bubble.js
"use strict"

var bubble_sort = function(array) {
    var n = array.length,
        intercambiado = true,
        temp;
    while (intercambiado) {
        intercambiado = false;
        for (var i = 1; i < n; i++) {
            //console.log(array[i - 1])
            if (array[i - 1] > array[i]) {
                temp = array[i - 1];
                array[i - 1] = array[i];
                array[i] = temp;
                intercambiado = true;
            }
        }
    }
}

if (require.main == module){
    var array = [1, 5, 65, 23, 57, 1232, -1, -5, -2, 242, 100,
                 4, 423, 2, 564, 9, 0, 10, 43, 64, 32, 1, 999]
    console.log(array.toString());
    bubble_sort(array)
    console.log(array.toString())
}

File: backend/src/algoritmos/sort/bubble_sort/bubble.c
#include <stdio.h>

/**
 * Ordena un array de enteros largo
 * según el algoritmo de ordenación de burbuja
 * @param seq Array de números enteros
 * @param n   Número de elementos en el array
 */
void bubble_sort(long seq[], unsigned long n) {
    char intercambiado = 1;
    int i, temp;
    while (intercambiado == 1) {
        intercambiado = 0;
        for (i=1; i<=n; i++) {
            if (seq[i-1] > seq[i]) {
                temp = seq[i];
                seq[i] = seq[i-1];
                seq[i-1] = temp;
                intercambiado = 1;
            }
        }
    }
}

int main() {
    long array[11] = {1, 5, 65, 23, 57, 1232, -1, -5, -2, 242, 100};

    printf("Array desordenado: ");
    for (int i=0; i < 11; i++) {
        printf("%d ", array[i]);
    }
    printf("\n");

    bubble_sort(array, 11);

    printf("Array ordenado: ");
    for (int i=0; i < 11; i++) {
        printf("%d ", array[i]);
    }
    printf("\n");

    return 0;
}